silkworm/silkrpc/test/api_test_database.hpp

```diff
@@ -0,0 +1,280 @@
+/*
+   Copyright 2023 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <bit>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+#include <boost/asio.hpp>
+#include <boost/asio/co_spawn.hpp>
+#include <boost/asio/thread_pool.hpp>
+#include <nlohmann/json.hpp>
+
+#include <silkworm/core/chain/genesis.hpp>
+#include <silkworm/core/execution/address.hpp>
+#include <silkworm/core/execution/execution.hpp>
+#include <silkworm/core/state/in_memory_state.hpp>
+#include <silkworm/core/types/block.hpp>
+#include <silkworm/core/types/receipt.hpp>
+#include <silkworm/infra/common/directories.hpp>
+#include <silkworm/infra/test_util/log.hpp>
+#include <silkworm/node/db/access_layer.hpp>
+#include <silkworm/node/db/buffer.hpp>
+#include <silkworm/node/db/genesis.hpp>
+#include <silkworm/silkrpc/common/constants.hpp>
+#include <silkworm/silkrpc/ethdb/file/local_database.hpp>
+#include <silkworm/silkrpc/http/request_handler.hpp>
+#include <silkworm/silkrpc/test/context_test_base.hpp>
+
+namespace silkworm::rpc::test {
+
+std::filesystem::path get_tests_dir() {
+    auto working_dir = std::filesystem::current_path();
+
+    while (!std::filesystem::exists(working_dir / "third_party" / "execution-apis") && working_dir != "/") {
+        working_dir = working_dir.parent_path();
+    }
+
+    return working_dir / "third_party" / "execution-apis" / "tests";
+}
+
+mdbx::env_managed open_db(const std::string& chaindata_dir) {
+    db::EnvConfig chain_conf{
+        .path = chaindata_dir,
+        .create = true,
+        .exclusive = true,
+        .in_memory = true,
+        .shared = false};
+
+    return db::open_env(chain_conf);
+}
+
+InMemoryState populate_genesis(db::RWTxn& txn, const std::filesystem::path& tests_dir) {
```

Comment:
This `populate_genesis` function looks very similar if not identical to the same function in `rpc_api_test`, I think we have room for some good refactoring here. Next PR.